tests/test_georaster.py

```diff
@@ -616,6 +616,10 @@ def test_set_ndv(self) -> None:
         # Check that the number of no data value is correct
         assert np.count_nonzero(ndv_index.data) == 112088
 
+        # Check that nodata can also be set upon loading
+        r = gr.Raster(datasets.get_path("landsat_B4"), nodata=0)
```

Comment:
Is there not somewhere where the nodata will be automatically set to 0? Perhaps best to set it to something else, like 1, and then test it?

Reply:
Also just curious, what happens if a float nodataval is set on an int raster?

Reply:
Good point for the nodata value. Solved in 730c86d.

Reply:
Regarding your second question, apparently, it accepts it, but nothing happens as no value is equal to the nodata. 
I would say this is a reasonable behavior... what do you think?

Reply:
What happens for an integer raster when given `1.0` as a nodata value? Does it correctly register as `1`? I think it may be worth warning if a `float` ndv is given to an `int` raster.

Reply:
It works fine if one sets a float instead of an int:
```
import geoutils as gu
img = gu.Raster(gu.datasets.get_path("landsat_B4"))
print(np.min(img.data))
img.set_ndv(13.0)
print(np.min(img.data))
```
yields 13 then 14.
It seems like the value is rounded up when compared to an integer.
`img.set_ndv(12.9999); np.min(img.data)` -> 13
`img.set_ndv(13.9999); np.min(img.data)` -> 14

Reply:
Now an issue is raised when the nodata value is incompatible with self.dtypes. See f366f03.